packages/cli/api-importers/openapi/openapi-v2-importer/src/3.1/OpenAPIConverter.ts

```diff
@@ -98,6 +101,82 @@ export class OpenAPIConverter extends AbstractConverter<OpenAPIConverterContext3
         };
     }
 
+    private pruneIgnoredPaths({
+        context,
+        errorCollector
+    }: {
+        context: OpenAPIConverterContext3_1;
+        errorCollector: ErrorCollector;
+    }): void {
+        for (const [path, pathItem] of Object.entries(context.spec.paths ?? {})) {
+            if (pathItem == null) {
+                continue;
+            }
+            const shouldIgnore = new FernIgnoreExtension({
+                breadcrumbs: ["paths", path],
+                operation: pathItem
+            }).convert({ context, errorCollector });
+
+            if (shouldIgnore && context.spec.paths != null) {
+                delete context.spec.paths[path];
+                continue;
+            }
+
+            for (const [method, operation] of Object.entries(pathItem)) {
+                if (operation == null || typeof operation !== "object") {
+                    continue;
+                }
+                const shouldIgnore = new FernIgnoreExtension({
+                    breadcrumbs: ["paths", path, method],
+                    operation
+                }).convert({ context, errorCollector });
+
+                if (shouldIgnore && context.spec.paths != null) {
+                    delete context.spec.paths[path];
+                    continue;
+                }
+
+                const parameters = "parameters" in operation ? operation.parameters : undefined;
+                const parametersToRemove: number[] = [];
+                for (const [index, parameter] of (parameters ?? []).entries()) {
+                    if (parameter == null) {
+                        continue;
+                    }
+                    const shouldIgnore = new FernIgnoreExtension({
+                        breadcrumbs: ["paths", path, method, "parameters"],
+                        operation: parameter
+                    }).convert({ context, errorCollector });
+
+                    if (shouldIgnore) {
+                        parametersToRemove.push(index);
+                    }
+                }
+
+                if (parameters != null) {
+                    for (const index of parametersToRemove.reverse()) {
+                        parameters.splice(index, 1);
+                    }
+                }
+            }
+        }
+
+        for (const [id, schema] of Object.entries(context.spec.components?.schemas ?? {})) {
+            if (schema == null) {
+                continue;
+            }
+            const shouldIgnore = new FernIgnoreExtension({
+                breadcrumbs: ["components", "schemas", id],
+                operation: schema
+            }).convert({ context, errorCollector });
+
+            if (shouldIgnore && context.spec.components != null) {
+                if (context.spec.components.schemas != null) {
+                    delete context.spec.components.schemas[id];
+                }
+            }
+        }
+    }
```

Comment:
@eyw520 thoughts on updating this to be more generic, imagine you just keep iterating over every key, value and wherever the key is `x-fern-ignore: true` you just delete the object? 

That way the extension can literally be used anywhere in the spec